### PR TITLE
executor: fix analyze update panic cause by duplicate call analyze executor Close method

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Close https://github.com/pingcap/tidb/issues/20389

related PR: https://github.com/pingcap/tidb/pull/19106

### Related changes

- No.

### Check List 

Tests 

- Unit test

Side effects

- No

### Release note 

- No

### CODE REVIEW

**Comment by crazycs520** [comment]
@SunRunAway PTAL

File: executor/explain.go [hunk 1]
```diff
@@ -73,14 +73,14 @@ func (e *ExplainExec) Next(ctx context.Context, req *chunk.Chunk) error {
...
-		}
-	}()
 	if e.analyzeExec != nil && !e.executed {
+		closed := false
+		defer func() {
+			if !closed && e.analyzeExec != nil {
+				err = e.analyzeExec.Close()
+				closed = true
+			}
+		}()
```
Comment (SunRunAway) [inline executor/explain.go:83]: This `defer` is duplicate to line 93. I think they can be merged into `defer` part and the `closed` variable can also be eliminated.

Comment (crazycs520) [reply executor/explain.go:83]: done.

Comment (crazycs520) [reply executor/explain.go:83]: done.

Comment (crazycs520) [reply executor/explain.go:83]: No, it will fail in test `TestApplyCache` and `TestExplainAnalyze`.

**Review by SunRunAway — COMMENTED** [review]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Review by tiancaiamao — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**